lib/buffer.js

```diff
@@ -36,7 +36,10 @@ function Buffer(arg) {
     // If less than zero, or NaN.
     if (arg < 0 || arg !== arg)
       arg = 0;
-    return allocate(arg);
+    const buffer = allocate(arg);
+    if (arguments[1])
```

Comment:
why not adding a argument as the below because `arguments` might have some problems from v8:

```
function Buffer(arg, zeroFill)
```